cou/commands.py

```diff
@@ -0,0 +1,343 @@
+        """Capitalize the help message for -h/--help.
+
+        :param action: group heading.
+        :type action: argparse.Action
+        """
+        if action.option_strings and (
+            "-h" in action.option_strings or "--help" in action.option_strings
+        ):
+            action.help = "Show this help message and exit."
+        super().add_argument(action)
+
+
+def get_subcommand_common_opts_parser() -> argparse.ArgumentParser:
+    """Create a shared parser for options specific to subcommands.
+
+    :return: a parser groups options commonly shared by subcommands
+    :rtype: argparse.ArgumentParser
+    """
+    # Define common options for subcommands and their children commands
+    subcommand_common_opts_parser = argparse.ArgumentParser(add_help=False)
+    subcommand_common_opts_parser.add_argument(
+        "--model",
+        default=None,
+        dest="model_name",
+        type=str,
+        help="Set the model to operate on. If unset, the model name will be determined by "
+        "inspecting the environment as follows:\n"
+        "  1 - Environment variable JUJU_MODEL,"
+        "  2 - Environment variable MODEL_NAME,"
+        "  3 - Current active juju model",
+    )
+    subcommand_common_opts_parser.add_argument(
+        "--parallel",
+        help="Run upgrade steps in parallel where possible.",
+        default=False,
+        action="store_true",
+    )
+
+    # quiet and verbose options are mutually exclusive
+    group = subcommand_common_opts_parser.add_mutually_exclusive_group()
+    group.add_argument(
+        "--verbose",
+        "-v",
+        default=0,
+        action="count",
+        dest="verbosity",
+        help="Increase logging verbosity in STDOUT. Multiple 'v's yield progressively "
+        "more detail (up to 4).\nNote that this doesn't affect the verbosity in logfile, "
+        "which will always have the maximum verbosity.",
+    )
+    group.add_argument(
+        "--quiet",
+        "-q",
+        default=False,
+        action="store_true",
+        dest="quiet",
+        help="Disable output in STDOUT.",
+    )
+
+    return subcommand_common_opts_parser
+
+
+def get_dataplane_common_opts_parser() -> argparse.ArgumentParser:
+    """Create a shared parser for options specific to data-plane.
+
+    :return: a parser groups options specific to data-plane
+    :rtype: argparse.ArgumentParser
+    """
+    # Define options specific to data-plane child command and make them mutually exclusive
+    dp_subparser = argparse.ArgumentParser(add_help=False)
+    dp_upgrade_group = dp_subparser.add_mutually_exclusive_group()
+    dp_upgrade_group.add_argument(
+        "--machine",
+        "-m",
+        action="append",
+        help="Specify machine id(s) to upgrade. This option accepts a single machine id as well "
+        "as a stringified comma-separated list for multiple machines. This option can be used "
+        "repetitively.",
```

Comment:
```suggestion
        help="Specify machine id(s) to upgrade. This option accepts a single machine id as well "
        "as a stringified comma-separated list of ids, and can be repeated multiple times.",
```

Reply:
and so on for `--hostname` and `--availability-zone`

Reply:
you've missed performing this change on `--hostname` and `--availability-zone`

Reply:
Done :)